### ISSUE TITLE
Unable to specify s3v2 when using MC_HOST_ env

### ISSUE THREAD

**Issue by mdsummers** [minio/mc#2915]
In the documentation there's a note about selecting s3v2 when using gcs.
```
NOTE: Google Cloud Storage only supports Legacy Signature Version 2, so you have to pick - S3v2
```
Following on from that the option to specify configuration for the session using `MC_HOST_` env is provided. I cannot see an option here to provide the API without going via `mc config` which is what I'm looking to avoid by using `MC_HOST_`.

## Expected behavior

I should be able to specify the api as S3v2 without using `mc config`, either by cli flag to existing commands, or via an env (making it consistent with `MC_HOST_`).

## Actual behavior

Specifying protocol as part of url (as indicated searching through issues/merge_requests) gives a parsing error:

Specifying with `https://` and using `--debug` appears to be using Authorization: AWS4-HMAC-SHA256, as opposed to just AWS when going via `mc config` and using the `--api` option.

## Steps to reproduce the behavior

Specifying api as part of protocol gives parsing error:
```
MC_HOST_s3=https+s3v2://ACCESS-KEY-HERE:SECRET-KEY-HERE@storage.googleapis.com mc ls s3/bucketname/
```

Using storage.googleapis.com defaults to s3v4:
```
MC_HOST_s3=https://ACCESS-KEY-HERE:SECRET-KEY-HERE@storage.googleapis.com mc ls s3/bucketname/ --debug
```

## mc version

```
Version: 2019-09-24T01:36:20Z
Release-tag: RELEASE.2019-09-24T01-36-20Z
Commit-id: 643835013047aa27ed35de0ac5a4ab9538f0cd68
```

## System information
```
Release-Tag:RELEASE.2019-09-24T01-36-20Z | Commit:643835013047 | Host:redacted | OS:linux | Arch:amd64 | Lang:go1.13 | Mem:6.8 MB/72 MB | Heap:6.8 MB/66 MB
```

**Comment by harshavardhana** [comment]
mc host is not meant for signature v2  btw are you sure about gcs ? I am able to access gcs using signature v4

**Comment by mdsummers** [comment]
I'm working on multi-client compat so I took the docs at face value. You're right, testing now I can see it's working fine on v4. 

> `NOTE: Google Cloud Storage only supports Legacy Signature Version 2, so you have to pick - S3v2`

This part of the docs is not accurate. Worth changing scope of this ticket to match that discrepancy? My implementation isn't blocked so happy to close. Please advise.

**Comment by harshavardhana** [comment]
Feel free to fix it @mdsummers - we are happy to accept PRs for doc changes.  If not close this issue we can fix it ourselves.

**Comment by mdsummers** [comment]
Looks like someone has beaten me to submitting changes.

Here was my take FWIW:
https://github.com/mdsummers/mc/commit/d3bbcd9d925de6cb32116d9a6b8086840490b349
https://github.com/mdsummers/mc/commit/a7d94d8e77f8ccde65578abe0b598ab8c8002cc5
https://github.com/mdsummers/mc/commit/a3b1e2970813b80c2a7db246dba7ceac29d404d8

Leaving this to be closed by MR or maintainer.

**Comment by harshavardhana** [comment]
Closed as fixed by #2916
